src/ui/server/controllers/insurance/your-buyer/outstanding-or-overdue-payments/index.ts

```diff
@@ -0,0 +1,220 @@
+    });
+  } catch (error) {
+    console.error('Error getting outstanding or overdue payments %O', error);
+
+    return res.redirect(PROBLEM_WITH_SERVICE);
+  }
+};
+
+/**
+ * post
+ * Check trading history validation errors and if successful, redirect to the next part of the flow.
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.redirect} Next part of the flow or error page
+ */
+export const post = async (req: Request, res: Response) => {
+  try {
+    const { application } = res.locals;
+
+    if (!application) {
+      return res.redirect(PROBLEM_WITH_SERVICE);
+    }
+
+    const {
+      referenceNumber,
+      buyer: { buyerTradingHistory },
+    } = application;
+
+    let isChange;
+    let isCheckAndChange;
+
+    /**
+     * If is a change route
+     * set isChange to true
+     */
+    if (isChangeRoute(req.originalUrl)) {
+      isChange = true;
+    }
+
+    /**
+     * If is a check-and-change route
+     * set isCheckAndChange to true
+     */
+    if (isCheckAndChangeRoute(req.originalUrl)) {
+      isCheckAndChange = true;
+    }
+
+    const payload = constructPayload(req.body, FIELD_IDS);
+
+    const validationErrors = generateValidationErrors(payload);
+
+    if (validationErrors) {
+      const { supportedCurrencies } = await api.keystone.APIM.getCurrencies();
+
+      if (!isPopulatedArray(supportedCurrencies)) {
+        return res.redirect(PROBLEM_WITH_SERVICE);
+      }
+
+      const generatedPageVariables = pageVariables(referenceNumber, supportedCurrencies, String(buyerTradingHistory[CURRENCY_CODE]));
+
+      return res.render(TEMPLATE, {
+        ...insuranceCorePageVariables({
+          PAGE_CONTENT_STRINGS,
+          BACK_LINK: req.headers.referer,
+        }),
+        ...generatedPageVariables,
+        userName: getUserNameFromSession(req.session.user),
+        validationErrors,
+        submittedValues: payload,
+      });
+    }
+
+    const saveResponse = await mapAndSave.buyerTradingHistory(payload, application);
+
+    if (!saveResponse) {
+      return res.redirect(PROBLEM_WITH_SERVICE);
+    }
+
+    const hasNoFailedPayments = buyerTradingHistory[FAILED_PAYMENTS] === null;
+
+    /**
+     * If is a change route
+     * redirect to CHECK_YOUR_ANSWERS
+     */
+    if (isChange) {
+      if (hasNoFailedPayments) {
+        return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${FAILED_TO_PAY_CHANGE}`);
+      }
+
+      return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${CHECK_YOUR_ANSWERS}`);
+    }
+
+    /**
+     * If is a check-and-change route
+     * redirect to CHECK_AND_CHANGE_ROUTE
+     */
+    if (isCheckAndChange) {
+      if (hasNoFailedPayments) {
+        return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${FAILED_TO_PAY_CHECK_AND_CHANGE}`);
+      }
+
+      return res.redirect(`${INSURANCE_ROOT}/${referenceNumber}${CHECK_AND_CHANGE_ROUTE}`);
+    }
+
```

Comment:
Can we update the documetation please for the conditions on line 194 and 206, it only mentions 1x redirect